daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -80,6 +83,28 @@ func init() {
 	}
 }
 
+func getStreamName(ctx logger.Context) (name string, err error) {
```

Comment:
Can you move this below `New`?  I'm not sure if the Docker project follows this style, but as a general rule I try to organize code such that the most abstract code is above more specific code, roughly that caller is above callee.  I think it makes code easier to read since you can get the general direction of how things work before diving into the details.  You can think of it as somewhat like a news article where the beginning is a summary and the details come below.